### DISCUSSION TITLE
에러 응답 코드 규칙

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/javascript-subway#32] [General]
# 현재 에러 상황 응답 코드 규칙

## 404 Not Found
- 리소스가 존재하지 않을 경우
- 사용자가 생성한 리소스가 아닐 경우 (해당 리소스에 접근할 수 없기 때문에 리소스가 존재하지 않다고 인식)

## 400 Bad Request
- 요청이 잘못되었을 경우
- 요청에 포함된 값이 없거나 유효하지 않을 경우
  - ex) 지하철 노선 생성 시 존재하지 않는 지하철 역 ID 를 포함했을 때
  - ex) 잘못된 비밀번호를 입력했을 때

## 422 Unprocessable Entity
- 요청 형식은 맞지만 처리할 수 없는 요청일 경우
  - ex) 이메일 중복 확인 시 이메일이 이미 존재하는 경우

### 500 Internal Server Error
- 서버에 오류가 발생하여 요청을 수행할 수 없을 경우
  - ex) 토큰 생성 실패 시

**Comment by gracefulBrown** [comment]
# 논의 해보고 싶은 이슈
## 이메일이 이미 존재하는 경우
- 기존 응답 코드
  - 회원 가입 시 이메일이 이미 존재하는 경우: 400
  - 이메일 중복 확인 시 이메일이 이미 존재하는 경우: 422

> 통일 하는게 좋을까요? 그러면 어떤 코드로 통일하는게 좋을까요?

**Reply by 365kim** [reply]
저는 422로 통일되는 것이 좋다고 생각합니다!
